<launch>
    <node pkg="ros_pololu_servo" type="ros_pololu_servo_node" name="ros_pololu_servo_node" output="screen">
        <param name="pololu_motors_yaml" value="$(find ros_pololu_servo)/launch/pololu_motors.yaml" />
        <param name="port_name" value="/dev/ttyACM0" />
        <param name="baud_rate" value="115200" />
        <param name="rate_hz" value="50" />
        <param name="daisy_chain" value="false" />
    </node>
    <!--
    <node pkg="race" type="race" name="race" output="screen">
        <param name="speed_test" value="0.55" />
    </node>
    <include file="$(find realsense2_camera)/launch/realsense2_camera.launch"/>
    -->

    <!-- steering pid -->
    <node name="heading" pkg="pid" type="controller" output="screen" >
        <param name="Kp" value="0.05" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.025" />
		<!--1575 speed: kp: .05, kd: .025-->
        <param name="upper_limit" value="45" />
        <param name="lower_limit" value="-45" />
        <param name="windup_limit" value="10" />
        <!--<param name="cutoff_frequency" value="20" />-->
        <param name="max_loop_frequency" value="10" />
        <param name="min_loop_frequency" value="5" />
        <param name="topic_from_plant" value="steering_plant" />
        <param name="setpoint_topic" value="steering_setpoint" />
        <param name="topic_from_controller" value="steering_effort" />
        <param name="pid_enable_topic" value="steering_pid_enable" />
    </node>
</launch>
